test/e2e/validation_test.go

```diff
@@ -289,13 +288,31 @@ func (tc *testContext) sshSetup() error {
 	return nil
 }
 
-// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
-// by the provided IP.
-func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+// runPowerShellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
+// PowerShell, on the host specified by the provided IP.
+func (tc *testContext) runPowerShellSSHJob(name, command, ip string) (string, error) {
+	keyMountDir := "/private-key"
+	sshCommand := []string{"bash", "-c",
+		fmt.Sprintf(
+			// first determine if the host has PowerShell or cmd as the default shell
+			"PSVERSION_TABLE=$(ssh -o StrictHostKeyChecking=no -i %s %s@%s 'echo ($PSVersionTable)');"+
+				"if [[ \"$PSVERSION_TABLE\" == *\"PSEdition\"* ]]; "+
+				"then CMD_PREFIX=\"\";CMD_SUFFIX=\"\";"+
+				// if PowerShell is not the default shell, explicitly run the command through PowerShell
+				"else CMD_PREFIX=\"powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \\\"\";CMD_SUFFIX=\"\\\"\";"+
```

Comment:
I think a constant `remotePowerShellCmdPrefix` already exists in the test suite holding the same value as `CMD_PREFIX` here. Might make this a bit cleaner.